doc: Replace remaining literal BTC with CURRENCY_UNIT.

domob1812 opened the pull request:
This replaces one remaining instance of the literal `"BTC"` string with the `CURRENCY_UNIT` constant, as is done in most of the codebase already.

After this change, no instance of literal `"BTC"` remains anywhere in the RPC help texts.

domob1812 committed:
Replace remaining literals BTC with CURRENCY_UNIT

This replaces one remaining instance of the literal "BTC" string with
the CURRENCY_UNIT constant, as is done in most of the codebase already.

The other remaining instance (which is just part of a log message and thus
not really user-visible) is just removed.

After this change, no instance of literal "BTC" remains anywhere in the
non-Qt and non-test codebase.

MarcoFalke commented on src/wallet/rpcwallet.cpp:
```diff
@@ -3361,7 +3361,7 @@ static UniValue bumpfee(const JSONRPCRequest& request)
                             {"fee_rate", RPCArg::Type::NUM, /* default */ "fallback to 'confTarget'", "FeeRate (NOT total fee) to pay, in " + CURRENCY_UNIT + " per kB\n"
             "                         Specify a fee rate instead of relying on the built-in fee estimator.\n"
-            "                         Must be at least 0.0001 BTC per kB higher than the current transaction fee rate.\n"},
+            "                         Must be at least 0.0001 " + CURRENCY_UNIT + " per kB higher than the current transaction fee rate.\n"},
```
```suggestion
                                     "Must be at least 0.0001 " + CURRENCY_UNIT + " per kB higher than the current transaction fee rate.\n"},
```

Can shorten the string now that it is touched for other reasons

domob1812 replied:
Makes sense, I've applied the change.  I'll squash everything into a single commit when review is otherwise done.

MarcoFalke commented:
I think this is ready to merge if it compiles

practicalswift commented:
Concept ACK

FWIW:

```
$ git grep BTC -- "*.cpp" "*.h" ":(exclude)src/qt/" ":(exclude)src/wallet/test/" ":(exclude)src/test/" | grep -vE ': *(//|\*) .*'
src/policy/feerate.cpp:const std::string CURRENCY_UNIT = "BTC";
src/validation.cpp:        LogPrint(BCLog::MEMPOOL, "replacing tx %s with %s for %s BTC additional fees, %d delta bytes\n",
src/wallet/rpcwallet.cpp:            "                         Must be at least 0.0001 BTC per kB higher than the current transaction fee rate.\n"},
```

domob1812 commented:
> Concept ACK
> 
> FWIW:
> 
> ```
> $ git grep BTC -- "*.cpp" "*.h" ":(exclude)src/qt/" ":(exclude)src/wallet/test/" ":(exclude)src/test/" | grep -vE ': *(//|\*) .*'
> src/policy/feerate.cpp:const std::string CURRENCY_UNIT = "BTC";
> src/validation.cpp:        LogPrint(BCLog::MEMPOOL, "replacing tx %s with %s for %s BTC additional fees, %d delta bytes\n",
> src/wallet/rpcwallet.cpp:            "                         Must be at least 0.0001 BTC per kB higher than the current transaction fee rate.\n"},
> ```

Good catch (I was only targeting the RPC help texts so far, as that is what I worked on when I noticed this).  I've updated the PR, replacing also the `validation.cpp` instance (and squashed the commits).

MarcoFalke commented on src/validation.cpp:
```diff
@@ -974,7 +974,7 @@ bool MemPoolAccept::Finalize(ATMPArgs& args, Workspace& ws)
     for (CTxMemPool::txiter it : allConflicting)
     {
-        LogPrint(BCLog::MEMPOOL, "replacing tx %s with %s for %s BTC additional fees, %d delta bytes\n",
+        LogPrint(BCLog::MEMPOOL, "replacing tx %s with %s for %s " + CURRENCY_UNIT + " additional fees, %d delta bytes\n",
```
```suggestion
        LogPrint(BCLog::MEMPOOL, "replacing tx %s with %s for %s additional fees, %d delta bytes\n",
```

I think we can just remove the unit here

domob1812 replied:
Good point, I agree as this is just a log message and not really user visible.  I've removed it here.

hebasto commented:
Concept ACK.

MarcoFalke commented:
ACK 7df0cf719fecf1b8d09801295db8bd2137bfe033

laanwj commented:
ACK 7df0cf719fecf1b8d09801295db8bd2137bfe033
Doesn't change any translation strings so can go into 0.20.0.
